### PR TITLE
feat: resource access control rfc

### PR DESCRIPTION
**Pull request by olevski**
RFC for: https://www.notion.so/Resource-Access-Controls-90a16ddf3b6240b2800f1612b9a52a0e?pvs=4

### CODE REVIEW

**Review by Panaetius — APPROVED** [review]

File: rfcs/011-resource-access-control/011-resource-access-control.md [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+to associate user sessions to resource pools.
+
+![Basic architecture diagram](./overall-flow.png)
+
+### The Resource Access Control Service
+
+The service exposes an HTTP API that the administrators interact with to create/update/delete resource pools and
+add/remove users to resource pools.
+
+A prototype of the API exposed by the service can be seen in the attached file: [api-prototype](./api-prototype.yaml)
```
Comment (rokroskar) [inline rfcs/011-resource-access-control/011-resource-access-control.md:65]: would it be possible to have an endpoint that returns all pools for a given user ID?

File: rfcs/011-resource-access-control/011-resource-access-control.md [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+administrator will take care of:
+ - (optional) Setting a label on the nodes dedicated to the resource pool
+ - Setting a taint on the nodes dedicated to the resource pool
+ - Specify toleration and (optionally) a label when creating/updating the resource pool associated with the nodes
+
+The above will result in three different modes of operation:
+
+1. No node taints/tolerations or affinities
+  - sessions can be scheduled on any default or untainted session node (subject to resource availability and node sizes)
+  - resource quotas still apply (based on priority class)
```
Comment (rokroskar) [inline rfcs/011-resource-access-control/011-resource-access-control.md:117]: what is a priority class? I didn't think we were defining them?

**Review by rokroskar — COMMENTED** [review]
Looks good, thanks @olevski and @aledegano - only obvious thing I found was the missing endpoint to return all resource pools for the user (will be needed by the UI).

File: rfcs/011-resource-access-control/011-resource-access-control.md (hunk 2 above)
Comment (olevski) [reply rfcs/011-resource-access-control/011-resource-access-control.md:117]: Priority class is a k8s resource. It allows us have quotas for resources that are assigned by k8s without having 1 k8s namespace per quota.

**Review by rokroskar — APPROVED** [review]
🚀

**Pull request merged**